Environmental Metagenomics Workshop
===================================

26 - 29 September 2017 at CICESE
--------------------------------
*Instructors: Harriet Alexander, Phil Brooks, and C. Titus Brown*

These are the online materials for the environmental metagenomics workshop run at [CICESE](http://www.cicese.edu.mx/).

We will be using HackMD to take collective notes throughout the course. If you wish to add to or follow along the [HackMD can be found here](https://hackmd.io/BwdgnArApgJgDANgLRQcAjEgLFARhJXGSJAJgGYYQFT0oAzYWoA=).

### Schedule
#### Tuesday, Day 1:
##### Morning
* [Welcome!](welcome.html)
* [Logging onto an instance through Amazon Web Services (AWS)](aws/boot.html)
* [Working on the Command Line](command-line.html)

##### Lunch

##### Afternoon
* [Command Line Blast](running-command-line-blast.html)
* Sourmash?

##### Homework
* Read the [Critical Assessment of Metagenome Interpretation (CAMI) Paper](http://www.biorxiv.org/content/biorxiv/early/2017/01/09/099127.full.pdf);
* Read [Hu et al. 2016](http://mbio.asm.org/content/7/1/e01669-15.abstract). This is the paper from which we pulled all of our sample data.

#### Wednesday, Day 2:
##### Morning
*  Presentation and discussion on types of workflows in metagenomics and introduction to the dataset we will be using
* [Evaluating your short-read data set quality](quality.html)

##### Lunch

##### Afternoon
* [Assembling your short read data set with MEGAHIT](assemble.html)
* [Evaluating the assembly](assembly-evaluation.html)
* [Mapping short reads to the assembly](mapping.html) *time permitting*

#### Thursday, Day 3:
##### Morning
* [Annotating your assembly with Prokka](prokka_tutorial.html)
* [Quantifying abundance across samples with Salmon](salmon_tutorial.html)

##### Lunch

##### Afternoon
* [Binning genomes out of your metagenome](binning.html)
* [Quickly searching and comparing your samples with sourmash](sourmash.html)

#### Friday, Day 4:
##### Morning
* [Putting it all together with Anvi'o](anvio.html)

##### Lunch

##### Afternoon

* [A brief discussion of workflows & repeatability](workflows.html)

Resources:

[SEQ Answers](http://seqanswers.com/)

[Biostars](https://www.biostars.org/)

[Data Carpentry](http://www.datacarpentry.org/)

[DIB Summer Institute](http://ivory.idyll.org/dibsi/)

### Undone

See [the complete table of contents](toc.html)

### Technical information

The github repository for this workshop is public at
https://github.com/ngs-docs/2017-ucsc-metagenomics
