<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Anvi’o &mdash; 2017-dibsi-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-dibsi-metagenomics 1.0 documentation" href="toc.html" />
    <link rel="next" title="Using and Installing Circos" href="circos_tutorial.html" />
    <link rel="prev" title="Slicing and dicing with k-mers" href="slice.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="toc.html">
          2017-dibsi-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="index.html">2017 DIBSI Metagenomics Workshop at UC Davis</a></li>
<li class="toctree-l1"><a class="reference internal" href="welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="jetstream/boot.html">Booting a Jetstream Computer Instance for your use!</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html">Intro to Shell Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#the-shell">The Shell</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#for-future-reference">For Future Reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#finding-files">Finding files</a></li>
<li class="toctree-l1"><a class="reference internal" href="running-command-line-blast.html">Running command-line BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1"><a class="reference internal" href="prokka_tutorial.html">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html">A sourmash tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html#k-mers-k-mer-specificity-and-comparing-samples-with-k-mer-jaccard-distance">K-mers, k-mer specificity, and comparing samples with k-mer Jaccard distance.</a></li>
<li class="toctree-l1"><a class="reference internal" href="binning.html">Binning a Metagenomic Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Anvi&#8217;o</a></li>
<li class="toctree-l1"><a class="reference internal" href="circos_tutorial.html">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="README.html">2017 / April / Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Anvi&#8217;o</a><ul>
<li><a class="reference internal" href="#using-anvi-o-to-knit-everything-together">Using Anvi&#8217;o to Knit Everything Together</a></li>
<li><a class="reference internal" href="#installing-anvi-o-and-a-few-other-programs">Installing anvi&#8217;o (and a few other programs)</a></li>
<li><a class="reference internal" href="#getting-it-into-anvi-o-format">Getting it into Anvi&#8217;o format</a></li>
<li><a class="reference internal" href="#mapping-data">Mapping data</a></li>
<li><a class="reference internal" href="#generating-contigs-database">Generating contigs database</a></li>
<li><a class="reference internal" href="#identifing-and-refining-genome-bins">Identifing and refining genome bins</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="slice.html" title="Previous Chapter: Slicing and dicing with k-mers"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Slicing and d...</span>
    </a>
  </li>
  <li>
    <a href="circos_tutorial.html" title="Next Chapter: Using and Installing Circos"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Using and Ins... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/anvio.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="anvi-o">
<span id="anvi-o"></span><h1>Anvi&#8217;o<a class="headerlink" href="#anvi-o" title="Permalink to this headline">¶</a></h1>
<div class="section" id="using-anvi-o-to-knit-everything-together">
<span id="using-anvi-o-to-knit-everything-together"></span><h2>Using Anvi&#8217;o to Knit Everything Together<a class="headerlink" href="#using-anvi-o-to-knit-everything-together" title="Permalink to this headline">¶</a></h2>
<p>Now we are going bring it all together and visuzlize our assembly with <a class="reference external" href="http://merenlab.org/software/anvio/">Anvi&#8217;o</a>. Anvi&#8217;o is a powerful and extensible tool that might be easily applied to pan-genomic analysis as well as metagenomic analysis. The anvi&#8217;o group has a series of fantstic online tutorials, including one on <a class="reference external" href="http://merenlab.org/2016/06/22/anvio-tutorial-v2/">metagenomic analysis</a>. They also run workshops periodically (<a class="reference external" href="http://merenlab.org/2016/08/18/events/">schedule here</a>) that throughly cover the use of the software.</p>
<p>Today, we are adapting their tutorial on metagenomic analysis to work with the subset dataset that we have.</p>
<p>The goals of this tutorial are to:</p>
<ul class="simple">
<li>Install anvi&#8217;o</li>
<li>Become familiar with the anvi&#8217;o workflow</li>
<li>Visualizing the assembly with anvi&#8217;o</li>
<li>Become familiar with the anvi&#8217;o interface</li>
<li>Learn how to refine and visuzlize genome bins with anvi&#8217;o</li>
</ul>
</div>
<div class="section" id="installing-anvi-o-and-a-few-other-programs">
<span id="installing-anvi-o-and-a-few-other-programs"></span><h2>Installing anvi&#8217;o (and a few other programs)<a class="headerlink" href="#installing-anvi-o-and-a-few-other-programs" title="Permalink to this headline">¶</a></h2>
<p>The first thing we need to do is install anvi&#8217;o. To install anvi&#8217;o we will be be using <a class="reference external" href="https://www.continuum.io/what-is-anaconda">Anaconda</a>.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~
wget https://repo.continuum.io/archive/Anaconda3-4.4.0-Linux-x86_64.sh
bash Anaconda3-4.4.0-Linux-x86_64.sh
</pre></div>
</div>
<p>Now, follow the prompts for the Anaconda installation. To finish the install it will ask you if you would like to add anaconda to the <code class="docutils literal"><span class="pre">$PATH</span></code> in you <code class="docutils literal"><span class="pre">.bashrc</span></code>. You should say yes. Now, you just need to source your <code class="docutils literal"><span class="pre">.bashrc</span></code> to make sure you can use conda.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="o">.</span><span class="n">bashrc</span>
</pre></div>
</div>
<p>Anaconda should now be installed. We will now use anaconda (<code class="docutils literal"><span class="pre">conda</span></code>) to install anvi&#8217;o (and all its dependencies) as well as source an environment in which to to run conda.</p>
<p>Now, install anvi&#8217;o using conda, create an environment in which to run it, and source the environment:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>conda create -n anvio232 -c bioconda -c conda-forge gsl anvio
source activate anvio232
</pre></div>
</div>
<p>Anvi&#8217;o should now be installed. But, let&#8217;s double check that it worked. They have a nice little test case to check that everything is working well as follows:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-self-test --suite mini
</pre></div>
</div>
<p>This prompt will start anvi&#8217;o processing and ultimately it will generate an interactive window with the anvi&#8217;o environment. This is accessible through port 8080 (typically, though it might create go to a different port that will be specified) at your IP address.</p>
<p>To find your IP address you can look back at the Jetstream instance page and copy the IP address.</p>
<p><img alt="IP-Address" src="_images/ip-address1.png" /></p>
<p>Now, open a new tab in your browser* and paste in the following:</p>
<p>*Note: This only works in Google Chrome.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>[Your IP Address]:8080
</pre></div>
</div>
<p>This should open up the anvi&#8217;o interface which is interactive and pretty good looking.</p>
<p>Now, we just need to install a few other programs, namely, samtools and Bowtie2, which we will use for mapping and looking at our mapped data.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>wget https://downloads.sourceforge.net/project/bowtie-bio/bowtie2/2.3.2/bowtie2-2.3.2-linux-x86_64.zip
unzip bowtie2-2.3.2-linux-x86_64.zip

echo &#39;export PATH=$PATH:~/bowtie2-2.3.2&#39; &gt;&gt; ~/.bashrc
source ~/.bashrc
sudo apt-get -y install samtools
</pre></div>
</div>
<p>Alright, now onto a complete re-analysis of our data with the anvi&#8217;o pipeline.</p>
</div>
<div class="section" id="getting-it-into-anvi-o-format">
<span id="getting-it-into-anvi-o-format"></span><h2>Getting it into Anvi&#8217;o format<a class="headerlink" href="#getting-it-into-anvi-o-format" title="Permalink to this headline">¶</a></h2>
<p>Anvi&#8217;o takes in 1) an assembly and 2) the raw read data. We have both of those already created, so let&#8217;s go ahead and download those data (trimmed reads and asssemblies):</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>mkdir ~/anvio-work
cd ~/anvio-work

curl -O https://s3-us-west-1.amazonaws.com/dib-training.ucdavis.edu/metagenomics-scripps-2016-10-12/SRR1976948.abundtrim.subset.pe.fq.gz
curl -O https://s3-us-west-1.amazonaws.com/dib-training.ucdavis.edu/metagenomics-scripps-2016-10-12/SRR1977249.abundtrim.subset.pe.fq.gz
curl -O https://s3-us-west-1.amazonaws.com/dib-training.ucdavis.edu/metagenomics-scripps-2016-10-12/subset_assembly.fa.gz
</pre></div>
</div>
<p>And, gunzip those files:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for file in *gz
    do
    gunzip $file
done
</pre></div>
</div>
<p>We now need to get our assembly into the correct format so that anvi&#8217;o interpret it.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-script-reformat-fasta subset_assembly.fa -o anvio-contigs.fa --min-len 2000 --simplify-names --report name_conversions.txt
</pre></div>
</div>
<p>Take a look at the output files. What has changed?</p>
</div>
<div class="section" id="mapping-data">
<span id="mapping-data"></span><h2>Mapping data<a class="headerlink" href="#mapping-data" title="Permalink to this headline">¶</a></h2>
<p>We need to map our reads to our anvi&#8217;o corrected assembly. This is going to take a little bit of time. First, build the an index for bowtie2:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>bowtie2-build anvio-contigs.fa anvio-contigs
</pre></div>
</div>
<p>We can write a for loop to map our two datasets and produce .bam files for the files:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for file in *fq
do
    bowtie2 --threads 8 -x anvio-contigs --interleaved $file -S ${file/.fq/}.sam
    samtools view -U 4 -bS ${file/.fq/}.sam &gt; ${file/.fq/}.bam
done
</pre></div>
</div>
<p>As above, we need to make these data readable for anvi&#8217;o:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for file in *.bam
do
    anvi-init-bam ${file} -o ${file/.bam/}.anvio.bam
done
</pre></div>
</div>
</div>
<div class="section" id="generating-contigs-database">
<span id="generating-contigs-database"></span><h2>Generating contigs database<a class="headerlink" href="#generating-contigs-database" title="Permalink to this headline">¶</a></h2>
<p>In this step we are asking anvi&#8217;o to create a database with information about your contigs. The contig database is fairly extensible and can contain lots of different information (taxonomic, functional, etc.). Here, we are primarily asking it to do three things:</p>
<ol class="simple">
<li>&#8216;Soft split&#8217; long contigs (&gt;20k): Anvi&#8217;o shows the generalized statistics for each contig (GC content, etc.). For long contigs these stats are calculated across split contigs (which remain grouped)</li>
<li>Identify and locate open reading frames in your contigs (using Prodigal)</li>
<li>Estimate Single Copy Gene content (using hmmer against defined gene sets for <a class="reference external" href="http://www.pnas.org/content/110/14/5540.full">bacteria</a> and <a class="reference external" href="https://www.nature.com/nature/journal/v499/n7459/full/nature12352.html">archaea</a>)</li>
<li>Calculate k-mer frequencies for the contigs in our assemblies</li>
</ol>
<p>So, run the following command to generate the database:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-gen-contigs-database -f anvio-contigs.fa -o anvio-contigs.db
</pre></div>
</div>
<p>Then, run this command to perform the hmm search and identify single copy gene content:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-run-hmms -c anvio-contigs.db --num-threads 28
</pre></div>
</div>
<p>Now, we can layer on the coverge information from our two samples:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for file in *.anvio.bam
do
    anvi-profile -i $file -c anvio-contigs.db -T 28

done
</pre></div>
</div>
<p>And finally, we run the merge step. This will pull all the information together and create a merged anvi&#8217;o profile. This step will also run <a class="reference external" href="https://github.com/BinPro/CONCOCT">CONCOCT</a> (<em>another</em> binning algorithm) that will identify bins in our data. Finally, this step calculates the hierarchical relationship betwewen our contigs based on a variety of parameters.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-merge *ANVIO_PROFILE/PROFILE.db -o MERGED-SAMPLES -c anvio-contigs.db --enforce-hierarchical-clustering
</pre></div>
</div>
<p>Now we can visualize our data!</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-interactive -p MERGED-SAMPLES/PROFILE.db
-c anvio-contigs.db
</pre></div>
</div>
</div>
<div class="section" id="identifing-and-refining-genome-bins">
<span id="identifing-and-refining-genome-bins"></span><h2>Identifing and refining genome bins<a class="headerlink" href="#identifing-and-refining-genome-bins" title="Permalink to this headline">¶</a></h2>
<p>First, let&#8217;s summarize the bin information for our data. This will produce a series of text-based output files detailing some statistics on our genome bins:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-summarize -p MERGED-SAMPLES/PROFILE.db -c anvio-contigs.db -o SAMPLES-SUMMARY -C CONCOCT
</pre></div>
</div>
<p>Take a look at the output in <code class="docutils literal"><span class="pre">SAMPLES-SUMMARY</span></code>. What does it report?</p>
<p>Now you can visuzlize those data in the anvi&#8217;o style by simply adding the -C flag to the previous anvi-interactive command:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>anvi-interactive -p MERGED-SAMPLES/PROFILE.db
-c anvio-contigs.db -C CONCOCT
</pre></div>
</div>
<p>Now, we can actually refine the genome bins using anvi&#8217;o. This allows us to use human intutition and pattern recognition to better identify contigs that should co-occur.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span> anvi-refine -p MERGED-SAMPLES/PROFILE.db -c anvio-contigs.db -b Bin_4 -C CONCOCT
</pre></div>
</div>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>