<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Binning a Metagenomic Assembly &mdash; 2017-dibsi-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-dibsi-metagenomics 1.0 documentation" href="toc.html" />
    <link rel="next" title="Gene Abundance Estimation with Salmon" href="salmon_tutorial.html" />
    <link rel="prev" title="A sourmash tutorial" href="sourmash.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="toc.html">
          2017-dibsi-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="index.html">2017 DIBSI Metagenomics Workshop at UC Davis</a></li>
<li class="toctree-l1"><a class="reference internal" href="welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="jetstream/boot.html">Booting a Jetstream Computer Instance for your use!</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html">Intro to Shell Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#the-shell">The Shell</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#for-future-reference">For Future Reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#finding-files">Finding files</a></li>
<li class="toctree-l1"><a class="reference internal" href="running-command-line-blast.html">Running command-line BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1"><a class="reference internal" href="prokka_tutorial.html">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html">A sourmash tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html#k-mers-k-mer-specificity-and-comparing-samples-with-k-mer-jaccard-distance">K-mers, k-mer specificity, and comparing samples with k-mer Jaccard distance.</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Binning a Metagenomic Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1"><a class="reference internal" href="anvio.html">Using Anvi&#8217;o to Knit Everything Together</a></li>
<li class="toctree-l1"><a class="reference internal" href="circos_tutorial.html">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="README.html">2017 / April / Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Binning a Metagenomic Assembly</a><ul>
<li><a class="reference internal" href="#installing-binners">Installing binners</a></li>
<li><a class="reference internal" href="#get-a-rough-count-of-the-total-number-of-reads-mapping">Get a rough count of the total number of reads mapping</a></li>
<li><a class="reference internal" href="#binning-1-maxbin">Binning 1 - MaxBin</a></li>
<li><a class="reference internal" href="#binning-2-metabat">Binning 2 - MetaBAT</a></li>
<li><a class="reference internal" href="#visualizing-the-bins">Visualizing the bins</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="sourmash.html" title="Previous Chapter: A sourmash tutorial"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; A sourmash tu...</span>
    </a>
  </li>
  <li>
    <a href="salmon_tutorial.html" title="Next Chapter: Gene Abundance Estimation with Salmon"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Gene Abundanc... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/binning.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="binning-a-metagenomic-assembly">
<span id="binning-a-metagenomic-assembly"></span><h1>Binning a Metagenomic Assembly<a class="headerlink" href="#binning-a-metagenomic-assembly" title="Permalink to this headline">¶</a></h1>
<p>A common approach following metagenome assembly is binning, a process by which assembled contigs are collected into groups or &#8216;bins&#8217; that might then be assigned some taxonomic affiliation. There are many different tools that can be used for binning (see <a class="reference external" href="http://biorxiv.org/content/early/2017/01/09/099127">CAMI review for more details</a>). Here, we will be using <a class="reference external" href="https://microbiomejournal.biomedcentral.com/articles/10.1186/2049-2618-2-26">MaxBin</a> and <a class="reference external" href="https://bitbucket.org/berkeleylab/metabat">MetaBAT</a>, which are both user friendly and highly cited. To use these binners, we will first need to map our data against the assembled metagenome using bwa and then estimate relative abundances by contig. We will then inspect the bins generated by MaxBin and MetaBAT using VizBin.</p>
<div class="section" id="installing-binners">
<span id="installing-binners"></span><h2>Installing binners<a class="headerlink" href="#installing-binners" title="Permalink to this headline">¶</a></h2>
<p>MaxBin</p>
<div class="highlight-python"><div class="highlight"><pre><span></span> cd
 curl  https://downloads.jbei.org/data/microbial_communities/MaxBin/getfile.php?MaxBin-2.2.2.tar.gz &gt; MaxBin-2.2.2.tar.gz
 tar xzvf MaxBin-2.2.2.tar.gz
 cd MaxBin-2.2.2
 cd src
 make
 cd ..
 ./autobuild_auxiliary
 export PATH=$PATH:~/MaxBin-2.2.2
</pre></div>
</div>
<p>MetaBAT</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd 
curl -L https://bitbucket.org/berkeleylab/metabat/downloads/metabat-static-binary-linux-x64_v0.32.4.tar.gz &gt; metabatv0.32.4.tar.gz
tar xvf metabatv0.32.4.tar.gz
</pre></div>
</div>
<p>Time to finally run the Binners!
<strong>Note</strong>: MaxBin can take a lot of time to run and bin your metagenome. As this is a workshop, we are doing three things that sacrifice <em>quality</em> for <em>speed</em>.</p>
<ol class="simple">
<li>We are only using 2 of the 6 datasets that were generated for the
this project. Most binning software, rely upon
many samples to accurately bin data. And, we have subsampled the
data to make it faster to proess.</li>
<li>We did quick and dirty contig estimation using samtools. If you use MaxBin, I would recommend allowing them to generate the abudance tables using Bowtie2. But, again, this will add to the run time.</li>
<li>We are limiting the number of iterations that are performed through
the MaxBin expectation-maximization algorithm (5 iterations instead of
50+). This will likely limit the quality of the bins we get
out. So, users beware and read <a class="reference external" href="https://downloads.jbei.org/data/microbial_communities/MaxBin/README.txt">the user&#8217;s manual</a>
before proceeding with your own data analysis.</li>
</ol>
</div>
<div class="section" id="get-a-rough-count-of-the-total-number-of-reads-mapping">
<span id="get-a-rough-count-of-the-total-number-of-reads-mapping"></span><h2>Get a rough count of the total number of reads mapping<a class="headerlink" href="#get-a-rough-count-of-the-total-number-of-reads-mapping" title="Permalink to this headline">¶</a></h2>
<p>Q. Why is the read counts per contigs not an accurate number?</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/mapping

for i in *sorted.bam
  do
    samtools idxstats $i &gt; $i.idxstats
    cut -f1,3 $i.idxstats &gt; $i.counts
  done
</pre></div>
</div>
</div>
<div class="section" id="binning-1-maxbin">
<span id="binning-1-maxbin"></span><h2>Binning 1 - MaxBin<a class="headerlink" href="#binning-1-maxbin" title="Permalink to this headline">¶</a></h2>
<p>&#8211;</p>
<p>Maxbin uses <strong>read coverage</strong> &amp; <strong>tetranucleotide frequencies</strong> for each contig, and <strong>marker gene counts</strong> for each bin</p>
<p>First, we will get a list of the count files that we have to pass to MaxBin</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>mkdir ~/mapping/binning
cd ~/mapping/binning
ls ../*counts &gt; abundance.list
</pre></div>
</div>
<p>Now, on to the actual binning</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>run_MaxBin.pl -contig ../subset_assembly.fa -abund_list abundance.list -max_iteration 5 -out mbin
</pre></div>
</div>
<p>This will generate a series of files. Take a look at the files generated. In particular you should see a series of *.fasta files preceeded by numbers. These are the different genome bins predicted by MaxBin.</p>
<p>Take a look at the mbin.summary file. What is shown?</p>
<p>Now, we are going to generate a concatenated file that contains all of our genome bins put together. We will change the fasta header name to include the bin number so that we can tell them apart later.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for file in mbin.*.fasta
  do 
    num=${file//[!0-9]/}
    sed -e &quot;/^&gt;/ s/$/ ${num}/&quot; mbin.$num.fasta &gt;&gt; maxbin_binned.concat.fasta
  done
</pre></div>
</div>
<p>And finally make an annotation file for visualization</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>echo label &gt; maxbin_annotation.list
grep &quot;&gt;&quot; maxbin_binned.concat.fasta |cut -f2 -d &#39; &#39;&gt;&gt; maxbin_annotation.list
</pre></div>
</div>
</div>
<div class="section" id="binning-2-metabat">
<span id="binning-2-metabat"></span><h2>Binning 2 - MetaBAT<a class="headerlink" href="#binning-2-metabat" title="Permalink to this headline">¶</a></h2>
<p>&#8211;</p>
<p>MetaBAT uses <strong>read coverage</strong>, <strong>coverage variance</strong>, &amp; <strong>tetranucleotide frequencies</strong> for each contig. This is done with a custom script</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/mapping
~/metabat/jgi_summarize_bam_contig_depths --outputDepth depth_var.txt *sorted.bam
</pre></div>
</div>
<p>Setup another &#8220;binning&#8221; directory for this tool&#8217;s results</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>mkdir ~/mapping/binning_metabat
cd ~/mapping/binning_metabat
</pre></div>
</div>
<p>Run MetaBAT script</p>
<p><em>Note that we are outputting info to a logfile</em></p>
<div class="highlight-python"><div class="highlight"><pre><span></span>~/metabat/metabat -i ../subset_assembly.fa -a ../depth_var.txt --verysensitive -o metabat -v &gt; log.txt
</pre></div>
</div>
<p>Make the .fasta file of all binned sequences</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for file in metabat.*.fa
  do
    num=${file//[!0-9]/} 
   sed -e &quot;/^&gt;/ s/$/ ${num}/&quot; metabat.$num.fa &gt;&gt; metabat_binned.concat.fasta
done 
</pre></div>
</div>
<p>Make an annotation file of the bin numbers for annotation in VizBin</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>echo label &gt; metabat_annotation.list
grep &quot;&gt;&quot; metabat_binned.concat.fasta |cut -f2 -d &#39; &#39;&gt;&gt; metabat_annotation.list
</pre></div>
</div>
</div>
<div class="section" id="visualizing-the-bins">
<span id="visualizing-the-bins"></span><h2>Visualizing the bins<a class="headerlink" href="#visualizing-the-bins" title="Permalink to this headline">¶</a></h2>
<p>Now that we have our binned data from both MetaBAT and MaxBin there are several different things we can do. One thing we might want to do is check the quality of the binning&#8211; a useful tool for this is <a class="reference external" href="http://ecogenomics.github.io/CheckM/">CheckM</a>. Today, for the sake of time, we will visualize the bins that we just generated using VizBin.</p>
<p>First, install VizBin::</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd
sudo apt-get install libatlas3-base libopenblas-base default-jre
curl -L https://github.com/claczny/VizBin/blob/master/VizBin-dist.jar?raw=true &gt; VizBin-dist.jar
</pre></div>
</div>
<p>VizBin can run in OSX or Linux but is very hard to install on Windows. To simplify things we are going to run VizBin in the desktop emulator through JetStream (which is ... a bit clunky). So, go back to the Jetstream and open up the web desktop simulator.</p>
<p><img alt="" src="_images/VizBin-OpenDesktop.png" /></p>
<p>Open the terminal through the desktop simulator and open VizBin:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>java -jar VizBin-dist.jar
</pre></div>
</div>
<p>This should prompt VizBin to open in another window. First we will look at the output of the MaxBin assembly. Click the choose button to open file browser to navigate to the binning folder (<code class="docutils literal"><span class="pre">~/mapping/binning</span></code>). There you will find the concatenated binned fasta file (<code class="docutils literal"><span class="pre">maxbin_binned.concat.fasta</span></code>). Upload this file and hit run.</p>
<p><img alt="" src="_images/VizBin-LoadFile.png" /></p>
<p>What do you see? Read up a bit on <a class="reference external" href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-014-0066-1">VizBin</a> to see how the visualization is generated.</p>
<p>Now, upload the maxbin_annotation.list file as an annotation file to VizBin. The annotation file contains the bin id for each of the contigs in the assembly that were binned.</p>
<p><img alt="" src="_images/VizBin-AddFiles.png" /></p>
<p>Now, do the same for MetaBat!</p>
<p>Compare the results of the two binning methods-</p>
<ul class="simple">
<li>How many bins were found?</li>
<li>How distinct are the bins?</li>
</ul>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>