<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>So you want to get some sequencing data out of NCBI? &mdash; 2017-dibsi-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-dibsi-metagenomics 1.0 documentation" href="toc.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="toc.html">
          2017-dibsi-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">2017 DIBSI Metagenomics Workshop at UC Davis</a></li>
<li class="toctree-l1"><a class="reference internal" href="welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="jetstream/boot.html">Booting a Jetstream Computer Instance for your use!</a></li>
<li class="toctree-l1"><a class="reference internal" href="running-command-line-blast.html">Running command-line BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1"><a class="reference internal" href="prokka_tutorial.html">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="binning.html">Mapping and Binning a Metagenome Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1"><a class="reference internal" href="circos_tutorial.html">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="README.html">2017 / April / Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">So you want to get some sequencing data out of NCBI?</a><ul>
<li><a class="reference internal" href="#the-challenge">The challenge</a></li>
<li><a class="reference internal" href="#what-is-an-api-and-how-does-it-relate-to-ncbi">What is an API and how does it relate to NCBI?</a></li>
<li><a class="reference internal" href="#automating-with-an-api">Automating with an API</a></li>
<li><a class="reference internal" href="#exercise-downloading-data">Exercise - Downloading data</a></li>
<li><a class="reference internal" href="#comment-on-genbank-files">Comment on Genbank files</a></li>
<li><a class="reference internal" href="#challenge">Challenge:</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/building-blast-database.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="so-you-want-to-get-some-sequencing-data-out-of-ncbi">
<h1>So you want to get some sequencing data out of NCBI?<a class="headerlink" href="#so-you-want-to-get-some-sequencing-data-out-of-ncbi" title="Permalink to this headline">¶</a></h1>
<p>Requirements:  You&#8217;ll need to start an Ubuntu EC2 instance and have root access.  The first part of this tutorial will be on your local computer and then we&#8217;ll move onto the EC2 instance.  Also, this tutorial assumes that someone has talked to you about paths and you know how to change directories and execute a program on a file.  If you get an error that a program or file does not exist, make sure you are in the right path.</p>
<p>First, let&#8217;s think about how these databases are structured.  I am going to create a database for folks to deposit whole genome sequences.  What kind of information am I going to store in this?  Many of you may be familiar with such a database, hosted by the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/">NCBI</a>.  The scripts that complement this tutorial can be downloaded with the following:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>git clone https://github.com/adina/scripts-for-ngs.git
</pre></div>
</div>
<p>Let&#8217;s come up with a list of things we&#8217;d like stored in this database and discuss some  of the challenges involved in database creation, management, and access.</p>
<div class="section" id="the-challenge">
<h2>The challenge<a class="headerlink" href="#the-challenge" title="Permalink to this headline">¶</a></h2>
<p>So, you&#8217;ve been given a list of genomes and been asked to create a phylogenetic tree of these genomes.   How big would this list be before you thought about hiring an undergraduate to download sequences?</p>
<p>Say the list is only 3 genomes:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">CP000962</span>
<span class="n">CP000967</span>
<span class="n">CP000975</span>
</pre></div>
</div>
<p>The following are some ways with which I&#8217;ve used to grab genome sequences:</p>
<ol class="arabic simple">
<li>Use the web portal and look up each FASTA</li>
<li>Use the <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/refseq/">FTP site</a>, find each genome, and download manually</li>
<li>Use the NCBI Web Services API to download the data</li>
</ol>
<p>Among these, I&#8217;m going to assume many of you are familiar with the first two.  This tutorial then is going to go through an example of the first approach and then focus on using APIs.</p>
</div>
<div class="section" id="what-is-an-api-and-how-does-it-relate-to-ncbi">
<h2>What is an API and how does it relate to NCBI?<a class="headerlink" href="#what-is-an-api-and-how-does-it-relate-to-ncbi" title="Permalink to this headline">¶</a></h2>
<p>Here&#8217;s some <a class="reference external" href="http://stackoverflow.com/questions/7440379/what-exactly-is-the-meaning-of-an-api">answers</a>, among which my favorite is &#8220;an interface through which you access someone else&#8217;s code or through which someone else&#8217;s code accesses yours &#8211; in effect the public methods and properties.&#8221;</p>
<p>The NCBI has a whole toolkit which they call <em>Entrez Programming Utilities</em> or <em>eutils</em> for short.  You can read all about it in the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25501/">documentation</a>.  There are a lot of things you can do to interface with all things NCBI, including publications, etc., but I am going to focus today on downloading sequencing data.</p>
<p>To do this, you&#8217;re going to be using one tool in <em>eutils</em>, called <em>efetch</em>.  There is a whole chapter devoted to <a class="reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch">efetch</a> &#8211; when I first started doing this kind of work, this documentation always broke my heart.  Its easier for me to just show you how to use it.</p>
<p>You can use the NCBI efetch utility on your web browser (as is true of many APIs).  Open a web browser, and try the following URL to download the nucleotide genome for CB00962:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=CP000962&amp;rettype=fasta&amp;retmode=text
</pre></div>
</div>
<p>You&#8217;ll note that a file downloaded on your computer.  Take a look at it.  You&#8217;ll note that it looks a lot like what you would see if you had searched for CP00962 on the NCBI search page.  Check it out <a class="reference external" href="http://www.ncbi.nlm.nih.gov/nuccore/CP000962">here</a>.</p>
<p>If I want to access other kinds of data associated with this genome.  For example, if I want the Genbank file as an output rather than a FASTA file, I would try the following command:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=CP000962&amp;rettype=gb&amp;retmode=text
</pre></div>
</div>
<p>Do you notice the difference in these two commands?  Let&#8217;s breakdown the command here:</p>
<ol class="arabic simple">
<li>&lt;<a class="reference external" href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi">http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi</a>?&gt;  This is command telling your computer program (or your browser) to talk to the NCBI API tool efetch.</li>
<li>&lt;db=nuccore&gt;  This command tells the NCBI API that you&#8217;d like it to look in this particular database for some data.  Other databases that the NCBI has available can be found <a class="reference external" href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi">here</a>.</li>
<li>&lt;id=CP000962&gt;  This command tells the NCBI API efetch the ID of the genome you want to find.</li>
<li>&lt;rettype=gb&amp;retmode=text&gt;  These two commands tells the NCBI how the data is returned.  You&#8217;ll note that in the two examples above this command varied slightly.  In the first, we asked for only the FASTA sequence, while in the second, we asked for the Genbank file.  Here&#8217;s some elusive documentation on where to find these <a class="reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/?report=objectonly">&#8220;return&#8221; objects</a>.</li>
</ol>
<p>Also, a useful command is also &lt;version=1&gt;.  There are different versions of sequences and some times that is useful.  For reproducibility, I try to specify versions in my queries, see these <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Class/MLACourse/Modules/Format/exercises/qa_accession_vs_gi.html">comments</a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Notice the &#8220;&amp;&#8221; that comes between each of these little commands, it is necessary and important.</p>
</div>
</div>
<div class="section" id="automating-with-an-api">
<h2>Automating with an API<a class="headerlink" href="#automating-with-an-api" title="Permalink to this headline">¶</a></h2>
<p>Ok, let&#8217;s think of automating this sort of query.</p>
<p>In the shell, you could run the same commands above with the addition of <em>curl</em> (a program to get information from remote sources) on your EC2 instance:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>curl &quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=CP000962&amp;rettype=fasta&amp;retmode=text&quot;
</pre></div>
</div>
<p>You&#8217;ll see it fly on to your screen.  Don&#8217;t panic - you can save it to a file with the redirect command &#8220;&gt;&#8221; and make it more useful.:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>curl &quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=CP000962&amp;rettype=fasta&amp;retmode=text&quot; &gt; CP000962.fa
</pre></div>
</div>
<p>You&#8217;ve now saved a query you&#8217;ve sent to the NCBI API to a file.  Now, you could now imagine writing a program where you made a list of IDs you want to download and put it in a for loop, <em>curling</em> each genome and saving it to a file.  The following is a <a class="reference external" href="https://github.com/adina/scripts-for-ngs/blob/master/fetch-genomes.py">script</a>.  Thanks to Jordan Fish who gave me the original version of this script before I even knew how and made it easy to use.</p>
<p>This script has some nifty documentation that you can see by trying to execute the script.  To see the documentation for this script:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>python fetch-genomes.py
</pre></div>
</div>
<p>You&#8217;ll see that you need to provide a list of IDs and a directory where you want to save the downloaded files.  What do you need to provide to this script?  The first thing is a file that contains a list of IDs (note that this is a required format, each ID on a new line) to fetch the data from NCBI.  Second, you need to name a directory where you want the program to put the files you fetch from the NCBI API.   Note that the lazy programmer who wrote this script requires you to identify a directory that does not currently exist.</p>
<p>So to use this script, two things are needed.  What are they?</p>
<p>To help out, I have provided a list of 50 IDs in a file called &#8220;interesting-genomes.txt.&#8221;  How can you tell there are 50 genomes in this file?  But I don&#8217;t want to just go wild and download all 50 at once.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You may want to run this on just a few of these IDs to begin with.  You can create a smaller list using the <em>head</em> command with the -n parameter in the shell.  For example, head -n 3 interesting-genomes.txt &gt; 3genomes.txt.</p>
</div>
<dl class="docutils">
<dt>To run the script::</dt>
<dd>python fetch-genomes.py 3genomes.txt 3genomes</dd>
<dt>Take a look at what happened, and when you&#8217;re ready to try it for more files you could try the following.  Note that the directory you name in this command is arbitrary and defined by you::</dt>
<dd>python fetch-genomes.py interesting-genomes.txt genbank-files</dd>
</dl>
<p>Let&#8217;s take a look inside this script.  The meat of this script uses the following code:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">url_template</span> <span class="o">=</span> <span class="s2">&quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&amp;id=</span><span class="si">%s</span><span class="s2">&amp;rettype=gb&amp;retmode=text&quot;</span>
</pre></div>
</div>
<p>You&#8217;ll see that the <em>id</em> here is a string character which is obtained from list of IDs contained in a separate file.  The rest of the script manages where the files are being placed and what they are named.  It also prints some output to the screen so you know its running.</p>
</div>
<div class="section" id="exercise-downloading-data">
<h2>Exercise - Downloading data<a class="headerlink" href="#exercise-downloading-data" title="Permalink to this headline">¶</a></h2>
<p>Try modifying the fetch_genomes.py script to download just the FASTA sequences of the genes.</p>
<p>Running this script should allow you to download genomes to your heart&#8217;s content.  But how do you grab specific genes from this data then?  Specifically, the challenge was to make a phylogenetic tree of sequences, so let&#8217;s target the conserved bacterial gene, <em>16S ribosomal RNA gene</em>.</p>
</div>
<div class="section" id="comment-on-genbank-files">
<h2>Comment on Genbank files<a class="headerlink" href="#comment-on-genbank-files" title="Permalink to this headline">¶</a></h2>
<p>Genbank files have a special structure to them.  You can look at it and figure it out for the most part, or read about it in detail <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html">here</a>.  To find out if your downloaded Genbank files contain 16S rRNA genes, I like to run the following command:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>grep 16S *gbk
</pre></div>
</div>
<p>This should look somewhat familiar from your shell lesson, but basically we&#8217;re looking for anylines that contain the character &#8220;16S&#8221; in any Genbank file we&#8217;ve downloaded.  Note that you&#8217;ll have to run this in the directory where you downloaded these files.</p>
<p>The structure of the Genbank file allows you to identify 16S genes.  For example,</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>rRNA        9258..10759
            /gene=&quot;rrs&quot;
            /locus_tag=&quot;CLK_3816&quot;
            /product=&quot;16S ribosomal RNA&quot;
            /db_xref=&quot;Pathema:CLK_3816&quot;
</pre></div>
</div>
<p>You could write code to find text like &#8216;rRNA&#8217; and &#8216;/product=&#8221;16S ribosomal RNA&#8221;&#8217;, grab the location of the gene, and then go to the FASTA file and grab these sequences.  To make things easy, there are existing packages to parse Genbank files.  I have the most experience with BioPython.  To begin with, let&#8217;s just use BioPython to help us with our program.</p>
<p>First, we&#8217;ll have to install BioPython on your instance and they&#8217;ve made that pretty easy:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo apt-get update
sudo apt-get install python-biopython
</pre></div>
</div>
<p>Fan Yang (Iowa State University) and I wrote a script to extract 16S rRNA sequences from Genbank files, <a class="reference external" href="https://github.com/adina/scripts-for-ngs/blob/master/parse-genbank.py">here</a>.  It basically searches for text strings in the Genbank structure that is appropriate for these particular genes.  You can read more about BioPython <a class="reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.html">here</a> and its Genbank parser <a class="reference external" href="http://biopython.org/DIST/docs/api/Bio.GenBank-module.html">here</a>.  In this script, we are looking for an &#8220;rRNA&#8221; feature and looking for specific text in its &#8220;/product&#8221; line.  If this is true, we go through the genome sequence and extract the coordinates of these genes, providing the specific gene sequence.</p>
<p>To run this script on the Genbank file for CP000962.  Note make sure you are in the right directory for both the program and the files:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>python parse-genbank.py genbank-files/CP000962.gbk &gt; genbank-files/CP000962.gbk.16S.fa
</pre></div>
</div>
<p>The resulting output file contains all 16S rRNA genes from the given Genbank file.</p>
<p>To run this for multiple files, I use a shell for loop:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>for x in genbank-files/*; do python parse-genbank.py $x &gt; $x.16S.fa; done
</pre></div>
</div>
<p>There are multiple ways to get this done &#8211; but this is how I like to do it.  Now, you can figure out how you like to do it.</p>
<p>And there you have it, you can now pretty much automatically grab 16S rRNA genes from any number of genomes in NCBI databases.</p>
</div>
<div class="section" id="challenge">
<h2>Challenge:<a class="headerlink" href="#challenge" title="Permalink to this headline">¶</a></h2>
<p>Find your favorite gene, download a database of it from NCBI, and find matching sequences from a sequencing dataset.</p>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>