<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Using and Installing Circos &mdash; 2017-cicese-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-cicese-metagenomics 1.0 documentation" href="toc.html" />
    <link rel="next" title="Workflow and repeatability discussion" href="workflow.html" />
    <link rel="prev" title="Using Anvi’o to Knit Everything Together" href="anvio.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="toc.html">
          2017-cicese-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="index.html">Environmental Metagenomics Workshop</a></li>
<li class="toctree-l1"><a class="reference internal" href="welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="aws/boot.html">Start an Amazon Web Services computer:</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html">Intro to Shell Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#the-shell">The Shell</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#for-future-reference">For Future Reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#finding-files">Finding files</a></li>
<li class="toctree-l1"><a class="reference internal" href="running-command-line-blast.html">Running command-line BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1"><a class="reference internal" href="prokka_tutorial.html">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html">A sourmash tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html#k-mers-k-mer-specificity-and-comparing-samples-with-k-mer-jaccard-distance">K-mers, k-mer specificity, and comparing samples with k-mer Jaccard distance.</a></li>
<li class="toctree-l1"><a class="reference internal" href="binning.html">Binning a Metagenomic Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1"><a class="reference internal" href="anvio.html">Using Anvi&#8217;o to Knit Everything Together</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="README.html">2017 / September / Environmental Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Using and Installing Circos</a><ul>
<li><a class="reference internal" href="#installing-circos">Installing Circos</a></li>
<li><a class="reference internal" href="#visualizing-gene-coverage-and-orientation">Visualizing Gene Coverage and Orientation</a></li>
<li><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="anvio.html" title="Previous Chapter: Using Anvi’o to Knit Everything Together"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Using Anvi’o ...</span>
    </a>
  </li>
  <li>
    <a href="workflow.html" title="Next Chapter: Workflow and repeatability discussion"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Workflow and ... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/circos_tutorial.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="using-and-installing-circos">
<h1>Using and Installing Circos<a class="headerlink" href="#using-and-installing-circos" title="Permalink to this headline">¶</a></h1>
<p>Circos is a powerful visualization tool that allows for the creation of circular graphics to display complex genomic data (e.g. genome comparisons). On top of the circular ideogram generated can be layered any number of graphical information (heatmaps, scatter plots, etc.).</p>
<p>The goals of this tutorial are to:</p>
<ul class="simple">
<li>Install circos on your Ubuntu system</li>
<li>Use Circos to visualize our metagenomic data</li>
</ul>
<p>Note: Beyond this brief crash course , circos is very well-documented and has a great series of <a class="reference external" href="http://circos.ca/documentation/tutorials/">tutorials</a> and <a class="reference external" href="http://circos.ca/documentation/course/">course</a> materials that are useful.</p>
<div class="section" id="installing-circos">
<h2>Installing Circos<a class="headerlink" href="#installing-circos" title="Permalink to this headline">¶</a></h2>
<p>You&#8217;ll need to install one additional ubuntu package, libgd:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo apt-get -y install libgd-perl
</pre></div>
</div>
<p>Make a directory called circos and navigate into it. There, we will download and extract the latest version of circos:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd
mkdir circos
cd circos
curl -O http://dib-training.ucdavis.edu.s3.amazonaws.com/metagenomics-scripps-2016-10-12/circos-0.69-3.tar.gz
tar -xvzf circos-0.69-3.tar.gz
</pre></div>
</div>
<p>Circos runs within Perl and as such does not need to be compiled to run. So, we can just add the location of circos to our path variable. (Alternatively, you can append this statement to the end of your <code class="docutils literal"><span class="pre">.bashrc</span></code> file.)</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>export PATH=~/circos/circos-0.69-3/bin:$PATH
</pre></div>
</div>
<p>Circos does, however, require quite a few additional perl modules to operate correctly. To see what modules are missing and need to be downloaded type the following:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">circos</span> <span class="o">-</span><span class="n">modules</span> <span class="o">&gt;</span> <span class="n">modules</span>
</pre></div>
</div>
<p>Now, to download all of these we will be using CPAN, a package manager for perl. We are going to pick out all the missing modules and then loop over those modules and download them using cpan.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>grep missing modules |cut -f13 -d &quot; &quot; &gt; missing_modules
for mod in $(cat missing_modules);
  do
  sudo cpan install $mod;
  done
</pre></div>
</div>
<p>This will take a while to run. When it is done check that you now have all modules downloaded by typing:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">circos</span> <span class="o">-</span><span class="n">modules</span>
</pre></div>
</div>
<p>If you got all &#8216;ok&#8217; then you are good to go!</p>
<p>And with that, circos should be up and ready to go. Run the example by navigating to the examples folder within the circos folder.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/circos/circos-0.69-3/example
bash run
</pre></div>
</div>
<p>This will take a little bit to run but should generate a file called <code class="docutils literal"><span class="pre">circos.png</span></code>.  Open it and you can get an idea of the huge variety of things that are possible with circos and a lot of patience. We will not be attempting anything that complex today, however.</p>
</div>
<div class="section" id="visualizing-gene-coverage-and-orientation">
<h2>Visualizing Gene Coverage and Orientation<a class="headerlink" href="#visualizing-gene-coverage-and-orientation" title="Permalink to this headline">¶</a></h2>
<p>First, let&#8217;s make a directory where we will be doing all of our work for plotting:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>mkdir ~/circos/plotting
cd ~/circos/plotting
</pre></div>
</div>
<p>Now, link in the <code class="docutils literal"><span class="pre">*gff</span></code> file output from prokka (which we will use to define the location of genes in each of our genomes), the genome assembly file <code class="docutils literal"><span class="pre">final.contigs.fa</span></code>, and the <code class="docutils literal"><span class="pre">SRR*counts</span></code> files that we generated with salmon:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>ln -fs ~/data/prokka_annotation/*gff .
ln -fs ~/data/final.contigs.fa .
ln -fs ~/quant/*counts .
</pre></div>
</div>
<p>We also need to grab a set of useful scripts and config files for this plotting exercise:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>curl -L -O https://github.com/ngs-docs/2016-metagenomics-sio/raw/master/circos-build.tar.gz
tar -xvzf circos-build.tar.gz
curl -L -O https://s3-us-west-1.amazonaws.com/dib-training.ucdavis.edu/metagenomics-scripps-2016-10-12/subset_assembly.fa.gz
gunzip subset_assembly.fa.gz
mv subset_assembly.fa final.contigs.fa
</pre></div>
</div>
<p>We are going to limit the data we are trying to visualize and get longest contigs from our assembly. We can do this using a script from the khmer package:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>extract-long-sequences.py  final.contigs.fa -l 24000 -o final.contigs.long.fa
cp ~/data/quant/*counts .
</pre></div>
</div>
<p>Next, we will run a script that processes the data from the the files that we just moved to create circos-acceptable files. This is really the crux of using circos: figuring out how to get your data into the correct format.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>python parse_data_for_circos.py
</pre></div>
</div>
<p>If you are interested&#8211; take a look at the script and the input files to see how these data were manipulated.</p>
<p>Circos operates off of three main types of files: 1) a config files that dictate the style and inputs to your circos plot, 2) a karyotype file that defines the size and layout of your &#8220;chromosomes&#8221;, and 3) any data files that  you call in your config file that detail attributes you want to plot.</p>
<p>The above script generated our karyotype file and four different data files. What are they? How are they oriented?</p>
<p>Now, we all that is left is actually running circos. Navigate into the circos-build directory and type <code class="docutils literal"><span class="pre">circos</span></code>:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd circos-build
circos
</pre></div>
</div>
<p>This command should generate an <code class="docutils literal"><span class="pre">circos.svg</span></code> and <code class="docutils literal"><span class="pre">circos.png</span></code>. Check out the <code class="docutils literal"><span class="pre">circos.png</span></code>!</p>
<p>Now, let&#8217;s take a look at the file that controls this crazy figure&#8211; <code class="docutils literal"><span class="pre">circos.config</span></code>.</p>
<p>Try changing a few parameters&#8211; colors, radius, size, to see what you can do. Again, if you are into this type of visualization, do check out the extensive <a class="reference external" href="http://circos.ca/documentation/tutorials/">tutorial</a>.</p>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://genome.cshlp.org/content/early/2009/06/15/gr.092759.109.abstract">http://genome.cshlp.org/content/early/2009/06/15/gr.092759.109.abstract</a></li>
<li><a class="reference external" href="http://circos.ca/documentation/tutorials">http://circos.ca/documentation/tutorials</a></li>
<li><a class="reference external" href="http://circos.ca/documentation/course/">http://circos.ca/documentation/course/</a></li>
</ul>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>