<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Annotation with Prokka &mdash; 2017-cicese-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-cicese-metagenomics 1.0 documentation" href="toc.html" />
    <link rel="next" title="A sourmash tutorial" href="sourmash.html" />
    <link rel="prev" title="Run the MEGAHIT assembler" href="assemble.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="toc.html">
          2017-cicese-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="index.html">Environmental Metagenomics Workshop</a></li>
<li class="toctree-l1"><a class="reference internal" href="welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="aws/boot.html">Start an Amazon Web Services computer:</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html">Intro to Shell Lesson</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#the-shell">The Shell</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#for-future-reference">For Future Reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="command-line.html#finding-files">Finding files</a></li>
<li class="toctree-l1"><a class="reference internal" href="running-command-line-blast.html">Running command-line BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html">A sourmash tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="sourmash.html#k-mers-k-mer-specificity-and-comparing-samples-with-k-mer-jaccard-distance">K-mers, k-mer specificity, and comparing samples with k-mer Jaccard distance.</a></li>
<li class="toctree-l1"><a class="reference internal" href="binning.html">Binning a Metagenomic Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1"><a class="reference internal" href="anvio.html">Using Anvi&#8217;o to Knit Everything Together</a></li>
<li class="toctree-l1"><a class="reference internal" href="circos_tutorial.html">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="README.html">2017 / September / Environmental Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Annotation with Prokka</a><ul>
<li><a class="reference internal" href="#installing-prokka">Installing Prokka</a></li>
<li><a class="reference internal" href="#running-prokka">Running Prokka</a></li>
<li><a class="reference internal" href="#questions">Questions?</a><ul>
<li><a class="reference internal" href="#alternative-annotation-tools-if-time-allows-or-for-homework">Alternative Annotation Tools (if Time Allows or for Homework)</a></li>
</ul>
</li>
<li><a class="reference internal" href="#install-kraken">Install Kraken</a></li>
<li><a class="reference internal" href="#install-kraken-mini-db">Install Kraken Mini DB</a></li>
<li><a class="reference internal" href="#running-kraken">Running Kraken</a></li>
<li><a class="reference internal" href="#install-prodigal">Install Prodigal</a></li>
<li><a class="reference internal" href="#running-prodigal">Running Prodigal</a></li>
<li><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="assemble.html" title="Previous Chapter: Run the MEGAHIT assembler"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Run the MEGAH...</span>
    </a>
  </li>
  <li>
    <a href="sourmash.html" title="Next Chapter: A sourmash tutorial"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">A sourmash tu... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/prokka_tutorial.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="annotation-with-prokka">
<h1>Annotation with Prokka<a class="headerlink" href="#annotation-with-prokka" title="Permalink to this headline">¶</a></h1>
<p>Prokka is a tool that facilitates the fast annotation of prokaryotic genomes.</p>
<p>The goals of this tutorial are to:</p>
<ul class="simple">
<li>Install Prokka</li>
<li>Use Prokka to annotate our genomes</li>
</ul>
<div class="section" id="installing-prokka">
<h2>Installing Prokka<a class="headerlink" href="#installing-prokka" title="Permalink to this headline">¶</a></h2>
<p>Download and extract the latest version of prokka:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/
git clone https://github.com/tseemann/prokka.git
</pre></div>
</div>
<p>We also will need some dependencies such as bioperl:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo apt-get -y install bioperl libdatetime-perl libxml-simple-perl libdigest-md5-perl
</pre></div>
</div>
<p>This may take a little while.</p>
<p>and we need an XML package from perl</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo bash
export PERL_MM_USE_DEFAULT=1
export PERL_EXTUTILS_AUTOINSTALL=&quot;--defaultdeps&quot;
perl -MCPAN -e &#39;install &quot;XML::Simple&quot;&#39;
exit
</pre></div>
</div>
<p>Now, you should be able to add Prokka to your <code class="docutils literal"><span class="pre">$PATH</span></code> and set up the index for the sequence database:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>export PATH=$PATH:$HOME/prokka/bin
prokka --setupdb
</pre></div>
</div>
<p>To make sure the database loaded directly:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">prokka</span> <span class="o">--</span><span class="n">listdb</span>
</pre></div>
</div>
<p>You should see something like:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>tx160085@js-157-212:~$ prokka --listdb
[17:04:15] Looking for databases in: /home/tx160085/prokka/bin/../db
[17:04:15] * Kingdoms: Archaea Bacteria Mitochondria Viruses
[17:04:15] * Genera: Enterococcus Escherichia Staphylococcus
[17:04:15] * HMMs: HAMAP
[17:04:15] * CMs: Bacteria Viruses
</pre></div>
</div>
<p>Prokka uses a core set of the Uniprot-DB Kingdom sets against which it blasts your samples.  It is possible to search in a more specific dataset, e.g. the genus Enterococcus, by adding a few flags to the command.</p>
<blockquote>
<div>&#8211;usegenus &#8211;genus Enterococcus</div></blockquote>
<p>Question:  What do you think you would do for adding to the default databases?</p>
<p>Prokka should be good to go now&#8211; you can check to make sure that all is well by typing <code class="docutils literal"><span class="pre">prokka</span></code>. This should print the help screen with all available options. You can find out more about Prokka databases <a class="reference external" href="https://github.com/tseemann/prokka#Databases">here</a>.</p>
</div>
<div class="section" id="running-prokka">
<h2>Running Prokka<a class="headerlink" href="#running-prokka" title="Permalink to this headline">¶</a></h2>
<p>Make a new directory for the annotation:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/
mkdir annotation
cd annotation
</pre></div>
</div>
<p>Link the metagenome assembly file into this directory:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>ln -fs ~/mapping/subset_assembly.fa .
</pre></div>
</div>
<p>Now it is time to run Prokka! There are tons of different ways to specialize the running of Prokka. We are going to keep it simple for now, though. It will take a little bit to run.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>prokka subset_assembly.fa --outdir prokka_annotation --prefix metagG --metagenome --kingdom Bacteria
</pre></div>
</div>
<p>Question:  Look at the results of the prokka analysis as it prepares your output file.  What types of categories are you seeing flash by on the screen?</p>
<p>Don&#8217;t worry, the program tends to pause here:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>Running: cat prokka_annotation\/sprot\.faa | parallel --gnu --plain -j 6 --block 242000
--recstart &#39;&gt;&#39; --pipe blastp -query --db /home/tx160085/prokka/bin/../db/kingdom/Bacteria/sprot
-evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no &gt; prokka_annotation\/sprot\.blast 2&gt; /dev/null
</pre></div>
</div>
<p>This will generate a new folder called <code class="docutils literal"><span class="pre">prokka_annotation</span></code> in which will be a series of files, which are detailed <a class="reference external" href="https://github.com/tseemann/prokka/blob/master/README.md#output-files">here</a>.</p>
<p>In particular, we will be using the <code class="docutils literal"><span class="pre">*.ffn</span></code> file to assess the relative read coverage within our metagenomes across the predicted genomic regions.</p>
<p>Question:  Take a moment and look inside the output files.:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/annotation/prokka_annotation
less -S *.fsa
</pre></div>
</div>
<p>less reminders:</p>
<p><a href="#id1"><span class="problematic" id="id2">*</span></a>Press space_bar to page down
<a href="#id3"><span class="problematic" id="id4">*</span></a>Press q to exit the less commands</p>
</div>
<div class="section" id="questions">
<h2>Questions?<a class="headerlink" href="#questions" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>What can I annotate with prokka?</li>
<li>Alternatives?</li>
<li>How do I submit my annotated files to <a class="reference external" href="https://github.com/tseemann/prokka/blob/master/README.md#NCBIGenbanksubmitter">Genbank? EBI?</a>?</li>
<li>Why is it called Prokka?</li>
</ul>
<div class="section" id="alternative-annotation-tools-if-time-allows-or-for-homework">
<h3>Alternative Annotation Tools (if Time Allows or for Homework)<a class="headerlink" href="#alternative-annotation-tools-if-time-allows-or-for-homework" title="Permalink to this headline">¶</a></h3>
<p>Kraken is a system for assigning taxonomic labels to short DNA sequences, usually obtained through metagenomic studies. Kraken aims to achieve high sensitivity and high speed by utilizing exact alignments of k-mers and a novel classification algorithm.  See <a class="reference external" href="https://ccb.jhu.edu/software/kraken/">Kraken Home Page</a> for more information.</p>
<p>Prodigal (Prokaryotic Dynamic Programming Genefinding Algorithm) is a microbial (bacterial and archaeal) gene finding program developed at Oak Ridge National Laboratory and the University of Tennessee. See the <a class="reference external" href="http://prodigal.ornl.gov">Prodigal home page</a> for more info.
<a class="reference external" href="http://denbi-metagenomics-workshop.readthedocs.io/en/latest/geneprediction/index.html">Citation</a></p>
<p>Prodigal is already installed inside the prokka wrapper, but sometimes it is handy to generate a standalone .gff file for annotation.</p>
</div>
</div>
<div class="section" id="install-kraken">
<h2>Install Kraken<a class="headerlink" href="#install-kraken" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~
git clone https://github.com/DerrickWood/kraken.git
cd ~/kraken
mkdir ~/kraken/bin
./install_kraken.sh ~/kraken/bin
export PATH=$PATH:$HOME/kraken/bin
</pre></div>
</div>
</div>
<div class="section" id="install-kraken-mini-db">
<h2>Install Kraken Mini DB<a class="headerlink" href="#install-kraken-mini-db" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python"><div class="highlight"><pre><span></span>mkdir ~/KRAKEN
cd ~/KRAKEN
wget http://ccb.jhu.edu/software/kraken/dl/minikraken.tgz
tar -xvf minikraken.tgz
</pre></div>
</div>
</div>
<div class="section" id="running-kraken">
<h2>Running Kraken<a class="headerlink" href="#running-kraken" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/annotation
mkdir kraken_annotation

kraken --db ~/KRAKEN/minikraken_20141208/ --threads 2 --fasta-input subset_assembly.fa --output kraken_annotation/subset_assembly.kraken


kraken-translate --db ~/KRAKEN/minikraken_20141208/  kraken_annotation/subset_assembly.kraken &gt; kraken_annotation/subset_assembly.kraken.labels
</pre></div>
</div>
<p>Kraken has now provided a taxonomic assignment to all of the clusters.</p>
<p>To generate a summary table:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/annotation
kraken-report --db ~/KRAKEN/minikraken_20141208 kraken_annotation/subset_assembly.kraken &gt; kraken_annotation/subset_assembly.kraken.report
</pre></div>
</div>
<p>The top of the file lists all the unclassified sequences, to look at the file and skip over these, do the following:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>grep -v ^U ~/annotation/kraken_annotation/subset_assembly.kraken.report | head -n20
</pre></div>
</div>
<p>The output of kraken-report is tab-delimited, with one line per taxon. The fields of the output, from left-to-right, are as follows:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Percentage of reads covered by the clade rooted at this taxon</li>
<li>Number of reads covered by the clade rooted at this taxon</li>
<li>Number of reads assigned directly to this taxon</li>
<li>A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply &#8216;-&#8216;.</li>
<li>NCBI taxonomy ID</li>
<li>indented scientific name</li>
</ol>
</div></blockquote>
<p>Example output:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>tx160085@js-157-212:~/annotation/kraken_annotation$ grep -v ^U subset_assembly.kraken.report | head -n20
 89.60  8311    8311    U       0       unclassified
 10.40  965     0       -       1       root
 10.40  965     3       -       131567    cellular organisms
 10.37  962     43      D       2           Bacteria
  6.51  604     0       P       200918        Thermotogae
  6.51  604     0       C       188708          Thermotogae
  6.51  604     0       O       2419              Thermotogales
  6.51  604     8       F       188709              Thermotogaceae
  5.11  474     0       G       28236                 Petrotoga
  5.11  474     0       S       69499                   Petrotoga mobilis
  5.11  474     474     -       403833                    Petrotoga mobilis SJ95
  1.22  113     0       G       1184396               Mesotoga
  1.22  113     0       S       1184387                 Mesotoga prima
  1.22  113     113     -       660470                    Mesotoga prima MesG1.Ag.4.2
  0.04  4       0       G       651456                Kosmotoga
  0.04  4       0       S       651457                  Kosmotoga olearia
  0.04  4       4       -       521045                    Kosmotoga olearia TBF 19.5.1
  0.02  2       1       G       2335                  Thermotoga
  0.01  1       0       S       177758                  Thermotoga lettingae
  0.01  1       1       -       416591                    Thermotoga lettingae TMO
</pre></div>
</div>
<p>Why use Kraken?</p>
<p>For a simulated metagenome of 100 bp reads in its fastest mode of operation, , Kraken processed over 4 million reads per minute on a single core, over 900 times faster than Megablast and over 11 times faster than the abundance estimation program MetaPhlAn. Kraken&#8217;s accuracy is comparable with Megablast, with slightly lower sensitivity and very high precision. <a class="reference external" href="http://denbi-metagenomics-workshop.readthedocs.io/en/latest/classification/kraken.html">Citation</a></p>
<p>However, kraken is only as sensitive as the provided database, so for unusual samples, a custom database needs to be constructed . The accuracy is very sensitive to the quantity of samples in the database.</p>
</div>
<div class="section" id="install-prodigal">
<h2>Install Prodigal<a class="headerlink" href="#install-prodigal" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~
wget https://github.com/hyattpd/Prodigal/releases/download/v2.6.3/prodigal.linux
tar -xvf v2.6.3.tar.gz
chmod 775 ~/prodigal.linux
</pre></div>
</div>
</div>
<div class="section" id="running-prodigal">
<h2>Running Prodigal<a class="headerlink" href="#running-prodigal" title="Permalink to this headline">¶</a></h2>
<p>Using prodigal with the same set of data, we can get a list of predicted genes.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cd ~/annotation
mkdir prodigal_annotation
~/prodigal.linux -p meta -a prodigal_annotation/subset_assembly.faa -d prodigal_annotation/subset_assembly.fna -f gff -o prodigal_annotation/subset_assembly.gff -i subset_assembly.fa
</pre></div>
</div>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://www.vicbioinformatics.com/software.prokka.shtml">http://www.vicbioinformatics.com/software.prokka.shtml</a></li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/24642063">https://www.ncbi.nlm.nih.gov/pubmed/24642063</a></li>
<li><a class="reference external" href="https://github.com/tseemann/prokka/blob/master/README.md">https://github.com/tseemann/prokka/blob/master/README.md</a></li>
<li><a class="reference external" href="http://denbi-metagenomics-workshop.readthedocs.io/en/latest/classification/kraken.html">http://denbi-metagenomics-workshop.readthedocs.io/en/latest/classification/kraken.html</a></li>
</ul>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>