<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Running command-line BLAST &mdash; 2017-dibsi-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-dibsi-metagenomics 1.0 documentation" href="toc.html" />
    <link rel="next" title="Short read quality and trimming" href="quality.html" />
    <link rel="prev" title="Booting a Jetstream Computer Instance for your use!" href="jetstream/boot.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="toc.html">
          2017-dibsi-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="index.html">2017 DIBSI Metagenomics Workshop at UC Davis</a></li>
<li class="toctree-l1"><a class="reference internal" href="welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="jetstream/boot.html">Booting a Jetstream Computer Instance for your use!</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Running command-line BLAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1"><a class="reference internal" href="prokka_tutorial.html">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="binning.html">Mapping and Binning a Metagenome Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1"><a class="reference internal" href="circos_tutorial.html">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="README.html">2017 / April / Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Running command-line BLAST</a><ul>
<li><a class="reference internal" href="#updating-the-software-on-the-machine">Updating the software on the machine</a></li>
<li><a class="reference internal" href="#running-blast">Running BLAST</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="jetstream/boot.html" title="Previous Chapter: Booting a Jetstream Computer Instance for your use!"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Booting a Jet...</span>
    </a>
  </li>
  <li>
    <a href="quality.html" title="Next Chapter: Short read quality and trimming"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Short read qu... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/running-command-line-blast.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="running-command-line-blast">
<span id="running-command-line-blast"></span><h1>Running command-line BLAST<a class="headerlink" href="#running-command-line-blast" title="Permalink to this headline">¶</a></h1>
<p>The goal of this tutorial is to run you through a demonstration of the
command line, which you may not have seen or used much before.</p>
<p><a class="reference external" href="jetstream/boot.html">Start up an m1.medium instance running Ubuntu 16.04 on Jetstream.</a></p>
<p>All of the commands below can copy/pasted.</p>
<div class="section" id="updating-the-software-on-the-machine">
<span id="updating-the-software-on-the-machine"></span><h2>Updating the software on the machine<a class="headerlink" href="#updating-the-software-on-the-machine" title="Permalink to this headline">¶</a></h2>
<p>Copy and paste the following commands:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo apt-get update &amp;&amp; sudo apt-get -y install python ncbi-blast+
</pre></div>
</div>
<p>(make sure to hit enter after the paste &#8211; sometimes the last line doesn&#8217;t
paste completely.)</p>
<p>This updates the software list and installs the Python programming
language and NCBI BLAST+.</p>
</div>
<div class="section" id="running-blast">
<span id="running-blast"></span><h2>Running BLAST<a class="headerlink" href="#running-blast" title="Permalink to this headline">¶</a></h2>
<p>First! We need some data.  Let&#8217;s grab the mouse and zebrafish RefSeq
protein data sets from NCBI, and put them in our home directory. If you&#8217;ve just logged
in, you should be there already, but if you&#8217;re unsure, just run <code class="docutils literal"><span class="pre">cd</span></code> and hit enter. Now,
we&#8217;ll use <code class="docutils literal"><span class="pre">curl</span></code> to download the files:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>curl -O ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.1.protein.faa.gz
curl -O ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.2.protein.faa.gz
curl -O ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.3.protein.faa.gz

curl -O ftp://ftp.ncbi.nih.gov/refseq/D_rerio/mRNA_Prot/zebrafish.1.protein.faa.gz
</pre></div>
</div>
<p>If you look at the files in the current directory, you should see four
files, along with a directory called lost+found which is for system
information:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="n">l</span>
</pre></div>
</div>
<p>should show you:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>total 29056
drwxr-xr-x 2 titus titus     4096 May  5 08:26 Desktop
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Documents
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Downloads
-rw-rw-r-- 1 titus titus  3610407 Jun 14 18:11 mouse.1.protein.faa.gz
-rw-rw-r-- 1 titus titus  6080985 Jun 14 18:11 mouse.2.protein.faa.gz
-rw-rw-r-- 1 titus titus  7520591 Jun 14 18:11 mouse.3.protein.faa.gz
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Music
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Pictures
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Public
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Templates
drwxr-xr-x 2 titus titus     4096 Jun 14 18:03 Videos
-rw-rw-r-- 1 titus titus 12500932 Jun 14 18:11 zebrafish.1.protein.faa.gz
</pre></div>
</div>
<p>All four of the files are FASTA protein files (that&#8217;s what the .faa
suggests) that are compressed with <code class="docutils literal"><span class="pre">gzip</span></code> (that&#8217;s what the .gz means).</p>
<p>Uncompress them:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>gunzip *.faa.gz
</pre></div>
</div>
<p>and let&#8217;s look at the first few sequences in the file:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>head mouse.1.protein.faa 
</pre></div>
</div>
<p>These are protein sequences in FASTA format.  FASTA format is something
many of you have probably seen in one form or another &#8211; it&#8217;s pretty
ubiquitous.  It&#8217;s a text file, containing records; each record
starts with a line beginning with a &#8216;&gt;&#8217;, and then contains one or more
lines of sequence text.</p>
<p>Let&#8217;s take those first two sequences and save them to a file.  We&#8217;ll
do this using output redirection with &#8216;&gt;&#8217;, which says &#8220;take
all the output and put it into this file here.&#8221;</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>head -11 mouse.1.protein.faa &gt; mm-first.fa
</pre></div>
</div>
<p>So now, for example, you can do <code class="docutils literal"><span class="pre">cat</span> <span class="pre">mm-first.fa</span></code> to see the contents of
that file (or <code class="docutils literal"><span class="pre">less</span> <span class="pre">mm-first.fa</span></code>).</p>
<p>Now let&#8217;s BLAST these two sequences against the entire zebrafish
protein data set. First, we need to tell BLAST that the zebrafish
sequences are (a) a database, and (b) a protein database.  That&#8217;s done
by calling &#8216;makeblastdb&#8217;:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>makeblastdb -in zebrafish.1.protein.faa -dbtype prot
</pre></div>
</div>
<p>Next, we call BLAST to do the search:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>blastp -query mm-first.fa -db zebrafish.1.protein.faa
</pre></div>
</div>
<p>This should run pretty quickly, but you&#8217;re going to get a lot of output!!
To save it to a file instead of watching it go past on the screen,
ask BLAST to save the output to a file that we&#8217;ll name <code class="docutils literal"><span class="pre">mm-first.x.zebrafish.txt</span></code>:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>blastp -query mm-first.fa -db zebrafish.1.protein.faa -out mm-first.x.zebrafish.txt
</pre></div>
</div>
<p>and then you can &#8216;page&#8217; through this file at your leisure by typing:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>less mm-first.x.zebrafish.txt
</pre></div>
</div>
<p>(Type spacebar to move down, and &#8216;q&#8217; to get out of paging mode.)</p>
<hr class="docutils" />
<p>Let&#8217;s do some more sequences (this one will take a little longer to run):</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>head -500 mouse.1.protein.faa &gt; mm-second.fa
blastp -query mm-second.fa -db zebrafish.1.protein.faa -out mm-second.x.zebrafish.txt
</pre></div>
</div>
<p>will compare the first 83 sequences.  You can look at the output file with:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>less mm-second.x.zebrafish.txt
</pre></div>
</div>
<p>(and again, type &#8216;q&#8217; to get out of paging mode.)</p>
<p>To get an output format that reads well into downstream applications, it is helpful to add the flag <em>-outfmt 6</em></p>
<div class="highlight-python"><div class="highlight"><pre><span></span>blastp -query mm-second.fa -db zebrafish.1.protein.faa -out mm-second.x.zebrafish.tbl.txt -outfmt 6
</pre></div>
</div>
<p>To see the results:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>head mm-second.x.zebrafish.tbl.txt | less -S
</pre></div>
</div>
<p>less -S means it is scrollable in screen from left to right.
type &#8216;q&#8217; to exit less</p>
<p>To find out how to customize blast outputs, it is helpful to look at the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK279668/">BLAST® Command Line Applications User Manual.</a></p>
<p>Notes:</p>
<ul>
<li><p class="first">you can execute multiple commands at a time;</p>
</li>
<li><p class="first">You might see a warning -</p>
<p><code class="docutils literal"><span class="pre">Selenocysteine</span> <span class="pre">(U)</span> <span class="pre">at</span> <span class="pre">position</span> <span class="pre">310</span> <span class="pre">replaced</span> <span class="pre">by</span> <span class="pre">X</span></code></p>
<p>what does this mean?</p>
</li>
<li><p class="first">why did it take longer to BLAST <code class="docutils literal"><span class="pre">mm-second.fa</span></code> than <code class="docutils literal"><span class="pre">mm-first.fa</span></code>?</p>
</li>
</ul>
<p>Things to mention and discuss:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">blastp</span></code> options and -help.</li>
<li>command line options, more generally - why so many?</li>
<li>automation rocks!</li>
</ul>
<p>Reminder: shut down your instance!</p>
<p>Other topics to discuss:</p>
<ul class="simple">
<li>when you shut down, you lose all your data</li>
<li>what computer(s) is this all happening on?</li>
</ul>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>