<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Running RStudio Server in the cloud &mdash; 2017-dibsi-metagenomics 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinxcontrib-images/LightBox2/lightbox2/css/lightbox.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="../_static/sphinxcontrib-images/LightBox2/lightbox2/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../_static/sphinxcontrib-images/LightBox2/lightbox2/js/lightbox.min.js"></script>
    <script type="text/javascript" src="../_static/sphinxcontrib-images/LightBox2/lightbox2-customize/jquery-noconflict.js"></script>
    <script type="text/javascript" src="../_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="../_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="../_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="2017-dibsi-metagenomics 1.0 documentation" href="../toc.html" />


<style type="text/css">
<!-- github-based editing disabled; set github_base_account
     and github_project -->
#editor-trap.toggled {
  display: none;
}
</style>



  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../toc.html">
          2017-dibsi-metagenomics</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../toc.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul>
<li class="toctree-l1"><a class="reference internal" href="../index.html">2017 Metagenomics workshop at UC Santa Cruz</a></li>
<li class="toctree-l1"><a class="reference internal" href="../welcome.html">Welcome!</a></li>
<li class="toctree-l1"><a class="reference internal" href="../jetstream/boot.html">Booting a Jetstream Computer Instance for your use!</a></li>
<li class="toctree-l1"><a class="reference internal" href="../quality.html">Short read quality and trimming</a></li>
<li class="toctree-l1"><a class="reference internal" href="../assemble.html">Run the MEGAHIT assembler</a></li>
<li class="toctree-l1"><a class="reference internal" href="../prokka_tutorial.html">Annotation with Prokka</a></li>
<li class="toctree-l1"><a class="reference internal" href="../binning.html">Mapping and Binning a Metagenome Assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="../salmon_tutorial.html">Gene Abundance Estimation with Salmon</a></li>
<li class="toctree-l1"><a class="reference internal" href="../mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../slice.html">Slicing and dicing with k-mers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../circos_tutorial.html">Using and Installing Circos</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflow.html">Workflow and repeatability discussion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../DATA.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../INSTALL.html">Building this site on your own</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html">2017 / April / Metagenomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../day2-install.html">Day 2 - installation instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kmer_trimming.html">K-mer Spectral Error Trimming</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Running RStudio Server in the cloud</a><ul>
<li><a class="reference internal" href="#start-up-an-amazon-instance">1. Start up an Amazon instance</a></li>
<li><a class="reference internal" href="#configure-your-network-firewall">2. Configure your network firewall</a></li>
<li><a class="reference internal" href="#log-in-via-the-shell">3. Log in via the shell</a></li>
<li><a class="reference internal" href="#set-a-password-for-the-ubuntu-account">4. Set a password for the &#8216;ubuntu&#8217; account</a></li>
<li><a class="reference internal" href="#install-r-and-the-gdebi-tool">5. Install R and the gdebi tool</a></li>
<li><a class="reference internal" href="#download-install-rstudio-server">6. Download &amp; install RStudio Server</a></li>
<li><a class="reference internal" href="#open-your-rstudio-server-instance">7. Open your RStudio Server instance</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="../_sources/aws/install-rstudio-server.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">

    <div class="col-md-12 content">
      
  
  <div class="section" id="running-rstudio-server-in-the-cloud">
<h1>Running RStudio Server in the cloud<a class="headerlink" href="#running-rstudio-server-in-the-cloud" title="Permalink to this headline">¶</a></h1>
<p>In this section, we will run RStudio Server on a remote Amazon machine.
This will require starting up an instance, configuring its network firewall,
and installing and running some software.</p>
<p>Reference documentation for running RStudio Server on Ubuntu:</p>
<blockquote>
<div><a class="reference external" href="https://www.rstudio.com/products/rstudio/download-server/">https://www.rstudio.com/products/rstudio/download-server/</a></div></blockquote>
<hr class="docutils" />
<div class="section" id="start-up-an-amazon-instance">
<h2>1. Start up an Amazon instance<a class="headerlink" href="#start-up-an-amazon-instance" title="Permalink to this headline">¶</a></h2>
<p>Start an ami-05384865 on an m4.xlarge machine, as per the instructions here:</p>
<p><a class="reference internal" href="boot.html"><em>Start an Amazon Web Services computer:</em></a>.</p>
</div>
<div class="section" id="configure-your-network-firewall">
<h2>2. Configure your network firewall<a class="headerlink" href="#configure-your-network-firewall" title="Permalink to this headline">¶</a></h2>
<p>Normally, Amazon computers only allow shell logins via ssh.
Since we want to run a Web service, we need to give the outside world
access to other network locations on the computer.</p>
<p>Follow these instructions:</p>
<p><a class="reference internal" href="configure-firewall.html"><em>Configure your instance firewall</em></a></p>
<p>(You can do this while the computer is booting.)</p>
</div>
<div class="section" id="log-in-via-the-shell">
<h2>3. Log in via the shell<a class="headerlink" href="#log-in-via-the-shell" title="Permalink to this headline">¶</a></h2>
<p>Follow these instructions to log in via the shell:</p>
<p><a class="reference internal" href="login-shell.html"><em>Log into your instance with the UNIX shell</em></a>.</p>
</div>
<div class="section" id="set-a-password-for-the-ubuntu-account">
<h2>4. Set a password for the &#8216;ubuntu&#8217; account<a class="headerlink" href="#set-a-password-for-the-ubuntu-account" title="Permalink to this headline">¶</a></h2>
<p>Amazon Web Services computers normally require a key (the .pem file)
instead of a login password, but RStudio Server will need us to log in
with a password.  So we need to configure a password for the account
we&#8217;re going to use (which is &#8216;ubuntu&#8217;)</p>
<p>Create a password like so:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo passwd ubuntu
</pre></div>
</div>
<p>and set it to something you&#8217;ll remember.</p>
</div>
<div class="section" id="install-r-and-the-gdebi-tool">
<h2>5. Install R and the gdebi tool<a class="headerlink" href="#install-r-and-the-gdebi-tool" title="Permalink to this headline">¶</a></h2>
<p>Update the software catalog and install a few things:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>sudo apt-get update &amp;&amp; sudo apt-get -y install gdebi-core r-base
</pre></div>
</div>
<p>This will take a few minutes.</p>
</div>
<div class="section" id="download-install-rstudio-server">
<h2>6. Download &amp; install RStudio Server<a class="headerlink" href="#download-install-rstudio-server" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python"><div class="highlight"><pre><span></span>wget https://download2.rstudio.org/rstudio-server-0.99.891-amd64.deb
sudo gdebi -n rstudio-server-0.99.891-amd64.deb
</pre></div>
</div>
<p>Upon success, you should see:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>Mar 07 15:20:18 ip-172-31-6-68 systemd[1]: Starting RStudio Server...
Mar 07 15:20:18 ip-172-31-6-68 systemd[1]: Started RStudio Server.
</pre></div>
</div>
</div>
<div class="section" id="open-your-rstudio-server-instance">
<h2>7. Open your RStudio Server instance<a class="headerlink" href="#open-your-rstudio-server-instance" title="Permalink to this headline">¶</a></h2>
<p>Finally, go to &#8216;<a class="reference external" href="http://">http://</a>&#8216; + your hostname + &#8216;:8787&#8217; in a browser,
eg.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>http://ec2-XX-YY-33-165.us-west-1.compute.amazonaws.com:8787/
</pre></div>
</div>
<p>and log into RStudio with username &#8216;ubuntu&#8217; and the password
you set it to above.</p>
<p>Voila!</p>
<hr class="docutils" />
<p>You can now just go ahead and use this, or you can &#8220;stop&#8221; it, or you
can freeze into an AMI for later use.</p>
<p>Note that on reboot, RStudio Server will start up again and all your files
will be there.</p>
<p>Go back to the index: <a class="reference internal" href="index.html"><em>Starting up an Amazon Web Services machine</em></a>.</p>
</div>
</div>



<hr>

<font size="-1"><b>LICENSE:</b>
This documentation and all textual/graphic site content is released
under <a href='http://creativecommons.org/publicdomain/zero/1.0/'>
Creative Commons - 0
(CC0)</a> -- <a href='https://github.com/ngs-docs/2017-ucsc-metagenomics'>fork @
github</a>. </font>

<hr>


<!-- disqus commenting disabled; set disqus_shortname -->

    </div>
    

  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Lab for Data Intensive Biology.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.5.<br/>
    </p>
  </div>
</footer>




<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');

</script>


  </body>
</html>